frontend/src/scenes/insights/Insight.tsx

```diff
@@ -84,6 +85,12 @@ export function Insight({ insightId }: { insightId: InsightShortId | 'new' }): J
         return <InsightSkeleton />
     }
 
+    const debouncedOnChange = useDebouncedCallback((insightMetadata) => {
+        if (canEditInsight && insightMode === ItemMode.Edit) {
```

Comment:
Is there a situation where `canEditInsight` is false but `insightMode` is `ItemMode.Edit`? I think this should be handled at logic level. For now I removed `canEditInsight` here as I pushed a fix to annoying autofocus relying on `insightMode`.